RecoTracker/CkfPattern/src/CkfTrackCandidateMakerBase.cc

```diff
@@ -423,16 +423,32 @@ namespace cms{
 
          viTotHits+=recHits.size();        
 
-	 LogDebug("CkfPattern") << "getting initial state.";
-	 const bool doBackFit = (!doSeedingRegionRebuilding) & (!reverseTrajectories);
-	 std::pair<TrajectoryStateOnSurface, const GeomDet*> && initState = theInitialState->innerState( *it , doBackFit);
-
-	 // temporary protection againt invalid initial states
-	 if ( !initState.first.isValid() || initState.second == nullptr || edm::isNotFinite(initState.first.globalPosition().x())) {
-	   //cout << "invalid innerState, will not make TrackCandidate" << endl;
-	   continue;
-	 }
-	 
+
+         LogDebug("CkfPattern") << "getting initial state.";
+         Trajectory trialTrajectory = (*it);
+         std::pair<TrajectoryStateOnSurface, const GeomDet*> initState;
+         bool failed = false;
+
+         do {
+           // Drop last hit if previous backFitter was not successful
+           if(failed) {
+             LogDebug("CkfPattern") << "removing last hit";
+             trialTrajectory.pop();
+             LogDebug("CkfPattern") << "hits remaining " << trialTrajectory.foundHits();
+           }
+
+           // Get inner state
+           const bool doBackFit = (!doSeedingRegionRebuilding) & (!reverseTrajectories);
```

Comment:
For clarity, & should be &&, otherwise someone might think bitwise AND is actually needed here instead of logical AND. But this issue doesn't justify changing the PR at this stage -- only fix it if making other changes, too.